casper/src/main/scala/coop/rchain/casper/Estimator.scala

```diff
@@ -182,10 +182,17 @@ object Estimator {
       blockDag: BlockDagRepresentation[F],
       scores: Map[BlockHash, Long]
   ): F[List[BlockHash]] =
-    blockDag.children(b).map(_.getOrElse(Set.empty[BlockHash]).filter(scores.contains)).map {
-      case c if c.nonEmpty => c.toList
-      case _               => List(b)
-    }
+    for {
+      children <- blockDag
+                   .children(b)
+                   .map(maybeChildren => maybeChildren.getOrElse(Set.empty[BlockHash]))
+      scoredChildren = children.filter(scores.contains)
+      result = if (scoredChildren.nonEmpty) {
+        scoredChildren.toList
+      } else {
+        List(b)
+      }
+    } yield result
```

Comment:
The map inside for doesn't help the readability. Neither does ```for {... result = computeResult(...)} yield result``` 

Something along these lines looks more readable
```scala 
blockDag
  .children(b)
  .map { maybeChildren =>
      val scoredChildren = 
                maybeChildren
                    .getOrElse(Set.empty[BlockHash]))
                    .filter(scores.contains)

       if (scoredChildren.nonEmpty) 
            scoredChildren.toList 
       else
            List(b) 
    }